capstone/src/main/java/com/google/sps/servlets/StudySetServlet.java

```diff
@@ -175,6 +189,181 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
+        academicTime,
+        courseName,
+        cards
+      )
+    ) {
+      System.err.println("Data fields are not completely filled in.");
+      return;
+    }
+
+    ServletContext servletContext = getServletContext();
+    DataSource pool = (DataSource) servletContext.getAttribute("my-pool");
+    if (pool == null) {
+      System.err.println(
+        "Connection to SQL database not working because servlet is not conntect to the database"
+      );
+      return;
+    }
+
+    createRow(
+      ownerID,
+      title,
+      subject,
+      description,
+      university,
+      professor,
+      academicTime,
+      courseName,
+      creationTimestamp,
+      pool,
+      cards
+    );
+
+    response.sendRedirect("/createStudySet.html");
+  }
+
+  private void createCard(
+    ArrayList<LinkedTreeMap<String, String>> cards,
+    String studyID,
+    Connection conn
+  )
+    throws SQLException {
+    PreparedStatement updateCardTable = conn.prepareStatement(
+      insertCardString(cards, insertCardStatement)
+    );
+    for (int i = 0; i < cards.size(); ++i) {
+      String frontText = cards.get(i).get("front");
+      String backText = cards.get(i).get("back");
+
+      updateCardTable.setInt((i * 3) + 1, Integer.parseInt(studyID));
+      updateCardTable.setString((i * 3) + 2, frontText);
+      updateCardTable.setString((i * 3) + 3, backText);
+    }
+
+    updateCardTable.execute();
+  }
+
+  private void createRow(
+    String ownerID,
+    String title,
+    String subject,
+    String description,
+    String university,
+    String professor,
+    String academicTime,
+    String courseName,
+    Timestamp creationTimestamp,
+    DataSource pool,
+    ArrayList<LinkedTreeMap<String, String>> cards
+  ) {
+    try (Connection conn = pool.getConnection()) {
+      PreparedStatement updateTable = conn.prepareStatement(
+        INSERT_STUDY_SET_STATEMENT
+      );
+
+      updateTable.setInt(1, Integer.parseInt(ownerID));
+      updateTable.setString(2, title);
+      updateTable.setString(3, subject);
+      updateTable.setString(4, description);
+      updateTable.setInt(5, Integer.parseInt(university));
+      updateTable.setString(6, professor);
+      updateTable.setString(7, academicTime);
+      updateTable.setString(8, courseName);
+      updateTable.setTimestamp(9, creationTimestamp);
+      updateTable.setTimestamp(10, creationTimestamp);
+
+      ResultSet resultStudySet = updateTable.executeQuery();
+
+      String studyID = "";
+      while (resultStudySet.next()) {
+        studyID = resultStudySet.getString("id");
+      }
+
+      createCard(cards, studyID, conn);
+    } catch (SQLException ex) {
+      throw new RuntimeException("Unable to verify Connection", ex);
+    }
+  }
+
+  private String insertCardString(
+    ArrayList<LinkedTreeMap<String, String>> cards,
+    String insertStatement
+  ) {
+    String fullInsertStatement = insertStatement;
+
+    for (int i = 0; i < cards.size(); ++i) {
```

Comment:
Pretty much every language has something like `String.join` for adding commas between bits of text. In this case, you can replace this loop with a one-liner: `String.join(", ", Collections.nCopies(cards.size(), "(?, ?, ?)"))`